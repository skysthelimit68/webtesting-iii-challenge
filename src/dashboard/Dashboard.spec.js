// Test away

import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/react/cleanup-after-each';

import Dashboard from './Dashboard';

it('matches snapshot', () => {
    const { container } = render (<Dashboard />);
    expect(container).toMatchSnapshot();
})

it('make sure 2 buttons exist and set to default lock gate and close gate when app starts', () => {
    const { getByText } = render(<Dashboard />);
    getByText(/lock gate/i);
    getByText(/close gate/i);
    expect(getByText('Lock Gate').disabled).toBeTruthy();
    expect(getByText('Close Gate').disabled).toBeFalsy();
})