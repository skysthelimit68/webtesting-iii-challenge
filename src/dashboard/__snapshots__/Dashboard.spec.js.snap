// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
      data-testid="lockStatus"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
      data-testid="closeStatus"
    >
      Open
    </div>
  </div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      data-testid="closeButton"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
      data-testid="lockButton"
    >
      Close Gate
    </button>
  </div>
</div>
`;
